src/plots/layout_attributes.js

```diff
@@ -234,6 +234,15 @@ module.exports = {
             'Defaults to `layout.uirevision`.'
         ].join(' ')
     },
+    selectionrevision: {
```

Comment:
Ha I see, `data[i].uirevision` also works for `selectedpoints` :ok_hand:

Reply:
> Ha I see, `data[i].uirevision` also works for `selectedpoints` 👌

Does it? I don't think so, at least I don't see it when I try that.

> What's the rationale here for putting `selectionrevision` in layout instead of in the individual traces?

I can certainly think of a use case for resetting `selectedpoints` but not resetting any of the other trace attributes - any time the basic meaning of the trace is unchanged but the point indices are altered, like if you had a trace showing annual data and you change it to monthly or you change the date range. And I guess you could imagine extending this argument to say `selectedpoints` could be invalidated for one trace but not for others, though it would create a weird UX situation in which the selected points in the trace where `selectedpoints` did NOT get reset get lost amid the sea of no-longer-deselected points in the trace where `selectedpoints` DID get reset. So between that and the fact that when the user creates a selection it's not a per-trace action, that's my argument for making `selectedrevision` a `layout` attribute. The app developer would give this attribute some value that depends on every input that can affect the identity of point indices in any trace.

Now there's still a question about what `uirevision` attributes the other trace attributes should link to. For the most part it seems like `uid` takes care of resetting these attributes (and note that if `uid` changes, even attributes linked to `uirevision` attributes `layout` will reset)... I linked `colorbar.(x|y)` to `editrevision` because this is more about the overall layout of the plot than about anything in the trace itself. And I made `visible: 'legendonly'` controlled by `legend.uirevision`, because (a) that's what controls `layout.hiddenlabels` for `pie` traces, and (b) the only use case I could think of for it, aside from those covered by `trace.uid`, is a "reset legend visibility" control (so `legend.uirevision` would just be set to the `nclicks` for this control).

That leaves `name`, `colorbar.title`, and `dimensions[i].constraintrange` (and presumably `dimensionorder` and `columnorder` when we get to those) controlled by `trace.uirevision`. I can't really think of a use case for this (again, aside from those covered by `trace.uid`), but it does seem like if there ever was one the attribute would belong in the trace, and I don't want to leave any attributes *only* under the control of the main `layout.uirevision` with no inheritance, since then to reset one of those attributes without resetting the whole rest of the plot would require setting *all* the other lower-level attributes to some constant value.

Reply:
> Does it? I don't think so, at least I don't see it when I try that.

Ah. I think I got that from:

https://github.com/plotly/plotly.js/blob/874a72aa86778ef223a65d8ce77b1af8b761fdd0/src/plots/attributes.js#L173-L174

> I can certainly think of a use case for resetting `selectedpoints` but not resetting any of the other trace attributes

> So between that and the fact that when the user creates a selection it's not a per-trace action, that's my argument for making `selectedrevision` a `layout` attribute.

You got me convinced. Thanks for writing your thought process down.

> but it does seem like if there ever was one the attribute would belong in the trace, and I don't want to leave any attributes only under the control of the main layout.uirevision with no inheritance

Good point. I have no problem leaving `trace.uirevision` in there even though it has (very) limited use cases.

Reply:
> Ah. I think I got that from:

Good catch, that was wrong, thanks! -> https://github.com/plotly/plotly.js/pull/3236/commits/5a27c0043e46342d73b8d986f9ce995fc48ead60